h/groups/views.py

```diff
@@ -102,10 +102,11 @@ def _read_group(request, group):
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
 
-    result = search.search(request, private=False, params={
-        "group": group.pubid, "limit": 1000})
-    annotations = [presenters.AnnotationHTMLPresenter(a)
-                   for a in result['rows']]
+    result = search.search(request,
+                           private=False,
+                           params={"group": group.pubid, "limit": 1000})
+    annotations = [models.Annotation(a) for a in result['rows']]
+    annotations = [presenters.AnnotationHTMLPresenter(a) for a in annotations]
```

Comment:
Such an embrace of mutable state makes me slightly uncomfortable, but it's probably the most clear way to do this.